{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 02 - Add hydrophobic-polar protein encoding\n",
    "\n",
    "Explore an additional protein coding in sourmash (hydrophillic/hydrophobic encoding)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sometimes we want to work with development versions of software tools. To keep these separate from the master versions, we need to make new conda environments to run them in.\n",
    "\n",
    "Because of how jupyter and conda interact, it's best to install software on the command line, and run jupyter notebook from within that conda environment. So execute the following on the command line to install sourmash with the hp encoding code and then launch a jupyter notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run this on the command line, prior to starting this notebook:\n",
    "```\n",
    "# First, make a folder to work in:\n",
    "cd ~/2019-fall-rotation-diblab # change into your rotation folder\n",
    "\n",
    "#now download the czbiohub version of sourmash (need changes in PR #758 https://github.com/dib-lab/sourmash/pull/758)\n",
    "git clone https://github.com/czbiohub/sourmash czbiohub_sourmash #drops the code into new \"czbiohub_sourmash\" folder\n",
    "\n",
    "cd czbiohub_sourmash\n",
    "conda create -y -c bioconda -c conda-forge -n sourmash-hp jupyter bam2fasta python=3.7 # new environment with python=3.7 and jupyter\n",
    "cd czbiohub_sourmash\n",
    "pip install -e '.' # do a developer install of the code in this folder\n",
    "\n",
    "# activate the environment\n",
    "conda activate sourmash-hp\n",
    "\n",
    "#check that the software was installed\n",
    "sourmash\n",
    "sourmash compute -h\n",
    "\n",
    "# now let's run a jupyter notebook from this environment:\n",
    "cd ../ # change back out to main rotation directory\n",
    "jupyter notebook\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: sourmash [--protein] [--no-protein] [--dayhoff] [--no-dayhoff] [--hp]\r\n",
      "                [--no-hp] [--dna] [--no-dna] [-q] [--input-is-protein]\r\n",
      "                [-k KSIZES] [-n NUM_HASHES] [--check-sequence] [-f]\r\n",
      "                [-o OUTPUT] [--singleton] [--merge MERGED] [--name-from-first]\r\n",
      "                [--input-is-10x] [--count-valid-reads COUNT_VALID_READS]\r\n",
      "                [--write-barcode-meta-csv WRITE_BARCODE_META_CSV]\r\n",
      "                [-p PROCESSES] [--save-fastas SAVE_FASTAS]\r\n",
      "                [--line-count LINE_COUNT] [--track-abundance]\r\n",
      "                [--scaled SCALED] [--seed SEED] [--randomize]\r\n",
      "                [--license LICENSE]\r\n",
      "                [--rename-10x-barcodes RENAME_10X_BARCODES]\r\n",
      "                [--barcodes-file BARCODES_FILE]\r\n",
      "                filenames [filenames ...]\r\n",
      "sourmash: error: the following arguments are required: filenames\r\n"
     ]
    }
   ],
   "source": [
    "# Now, let's check the sourmash compute options\n",
    "!sourmash compute -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You should see a new option available for sourmash compute: `--hp`. \n",
    "\n",
    "You previously ran:\n",
    "+ DNA k = 21, 31, 51; scaled = 2000\n",
    "+ RNA k = 21, 31, 51; scaled = 2000\n",
    "+ protein k = 7, 11, 17; scaled = 2000; no encoding\n",
    "+ protein k = 7, 11, 17; scaled = 2000; dayhoff encoding\n",
    "\n",
    "You should now be able to compute the following signatures:\n",
    "+ protein k = 7, 11, 17; scaled = 2000; hp encoding\n",
    "\n",
    "### Task: \n",
    "  - Build signatures with hp encoding and sourmash compare matrices, as before. How does hp encoding compare to dayhoff encoding, or signatures built with the full AA alphabet? What about the DNA/RNA level? Can you build a visualization to showcase these differences?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For example, the following generates hp signatures for the bacteroides dataset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000011065.1_ASM1106v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000011065.1_ASM1106v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000011065.1_ASM1106v1_protein.faa.gz 4824 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4825 sequences in 2018-test_datasets/bacteroides/protein/GCA_000011065.1_ASM1106v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00308 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000011065.1_ASM1106v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000157015.1_ASM15701v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000157015.1_ASM15701v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000157015.1_ASM15701v1_protein.faa.gz 4926 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4927 sequences in 2018-test_datasets/bacteroides/protein/GCA_000157015.1_ASM15701v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00171 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000157015.1_ASM15701v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000159855.2_Bact_sp_3_2_5_V2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000159855.2_Bact_sp_3_2_5_V2_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000159855.2_Bact_sp_3_2_5_V2_protein.faa.gz 4130 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4131 sequences in 2018-test_datasets/bacteroides/protein/GCA_000159855.2_Bact_sp_3_2_5_V2_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00189 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000159855.2_Bact_sp_3_2_5_V2_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000163055.2_ASM16305v2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000163055.2_ASM16305v2_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000163055.2_ASM16305v2_protein.faa.gz 1959 sequences\r",
      "\u001b[Kcalculated 6 signatures for 1960 sequences in 2018-test_datasets/bacteroides/protein/GCA_000163055.2_ASM16305v2_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00148 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000163055.2_ASM16305v2_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000224595.1_Prevotella_sp_C561_V1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000224595.1_Prevotella_sp_C561_V1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000224595.1_Prevotella_sp_C561_V1_protein.faa.gz 3401 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3402 sequences in 2018-test_datasets/bacteroides/protein/GCA_000224595.1_Prevotella_sp_C561_V1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00165 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000224595.1_Prevotella_sp_C561_V1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000273215.1_Bact_ovat_CL03T12C18_V1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000273215.1_Bact_ovat_CL03T12C18_V1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000273215.1_Bact_ovat_CL03T12C18_V1_protein.faa.gz 5523 sequences\r",
      "\u001b[Kcalculated 6 signatures for 5524 sequences in 2018-test_datasets/bacteroides/protein/GCA_000273215.1_Bact_ovat_CL03T12C18_V1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00171 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000273215.1_Bact_ovat_CL03T12C18_V1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000373705.1_ASM37370v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000373705.1_ASM37370v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000373705.1_ASM37370v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000403155.2_Bact_thet_dnLKV9_V1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000403155.2_Bact_thet_dnLKV9_V1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000403155.2_Bact_thet_dnLKV9_V1_protein.faa.gz 4798 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4799 sequences in 2018-test_datasets/bacteroides/protein/GCA_000403155.2_Bact_thet_dnLKV9_V1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00174 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000403155.2_Bact_thet_dnLKV9_V1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000431275.1_MGS755_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000431275.1_MGS755_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000431275.1_MGS755_protein.faa.gz 1935 sequences\r",
      "\u001b[Kcalculated 6 signatures for 1936 sequences in 2018-test_datasets/bacteroides/protein/GCA_000431275.1_MGS755_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00176 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000431275.1_MGS755_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000432495.1_MGS558_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000432495.1_MGS558_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000432495.1_MGS558_protein.faa.gz 3728 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3729 sequences in 2018-test_datasets/bacteroides/protein/GCA_000432495.1_MGS558_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00183 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000432495.1_MGS558_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000466425.1_ASM46642v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000466425.1_ASM46642v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000466425.1_ASM46642v1_protein.faa.gz 2777 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2778 sequences in 2018-test_datasets/bacteroides/protein/GCA_000466425.1_ASM46642v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00168 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000466425.1_ASM46642v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000495955.1_NUHP2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000495955.1_NUHP2_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000495955.1_NUHP2_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000501415.1_BorBurgIPT48_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000501415.1_BorBurgIPT48_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000501415.1_BorBurgIPT48_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000542015.1_Stap_aure_FVRH6074_V1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000542015.1_Stap_aure_FVRH6074_V1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000542015.1_Stap_aure_FVRH6074_V1_protein.faa.gz 2679 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2680 sequences in 2018-test_datasets/bacteroides/protein/GCA_000542015.1_Stap_aure_FVRH6074_V1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00332 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000542015.1_Stap_aure_FVRH6074_V1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000550025.1_Stap_aure_M1496_V1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000550025.1_Stap_aure_M1496_V1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000550025.1_Stap_aure_M1496_V1_protein.faa.gz 2660 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2661 sequences in 2018-test_datasets/bacteroides/protein/GCA_000550025.1_Stap_aure_M1496_V1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00189 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000550025.1_Stap_aure_M1496_V1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000598325.1_ASM59832v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000598325.1_ASM59832v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000598325.1_ASM59832v1_protein.faa.gz 4832 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4833 sequences in 2018-test_datasets/bacteroides/protein/GCA_000598325.1_ASM59832v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00161 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000598325.1_ASM59832v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000611675.1_ASM61167v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000611675.1_ASM61167v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000611675.1_ASM61167v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000614865.1_ASM61486v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000614865.1_ASM61486v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000614865.1_ASM61486v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000614895.1_ASM61489v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000614895.1_ASM61489v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000614895.1_ASM61489v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000626635.1_ASM62663v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000626635.1_ASM62663v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000626635.1_ASM62663v1_protein.faa.gz 3970 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3971 sequences in 2018-test_datasets/bacteroides/protein/GCA_000626635.1_ASM62663v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00153 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000626635.1_ASM62663v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000707425.1_CFSAN007430_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000707425.1_CFSAN007430_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000707425.1_CFSAN007430_01.0_protein.faa.gz 3662 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3663 sequences in 2018-test_datasets/bacteroides/protein/GCA_000707425.1_CFSAN007430_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00420 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000707425.1_CFSAN007430_01.0_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000759315.1_04_NF40_HMP9302v01_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000759315.1_04_NF40_HMP9302v01_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000759315.1_04_NF40_HMP9302v01_protein.faa.gz 2013 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2014 sequences in 2018-test_datasets/bacteroides/protein/GCA_000759315.1_04_NF40_HMP9302v01_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00149 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000759315.1_04_NF40_HMP9302v01_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000785025.1_ASM78502v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000785025.1_ASM78502v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000785025.1_ASM78502v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000814325.1_ASM81432v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000814325.1_ASM81432v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000814325.1_ASM81432v1_protein.faa.gz 3301 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3302 sequences in 2018-test_datasets/bacteroides/protein/GCA_000814325.1_ASM81432v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00174 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000814325.1_ASM81432v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000821925.1_34AVR_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000821925.1_34AVR_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_000821925.1_34AVR_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000938735.2_ASM93873v2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000938735.2_ASM93873v2_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000938735.2_ASM93873v2_protein.faa.gz 4792 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4793 sequences in 2018-test_datasets/bacteroides/protein/GCA_000938735.2_ASM93873v2_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00183 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000938735.2_ASM93873v2_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_000974365.1_ASM97436v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_000974365.1_ASM97436v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_000974365.1_ASM97436v1_protein.faa.gz 4410 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4411 sequences in 2018-test_datasets/bacteroides/protein/GCA_000974365.1_ASM97436v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00174 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_000974365.1_ASM97436v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001049535.1_3731_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001049535.1_3731_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001049535.1_3731_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001049555.1_7330_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001049555.1_7330_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001049555.1_7330_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001055755.1_ASM105575v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001055755.1_ASM105575v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001055755.1_ASM105575v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001304045.1_ASM130404v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001304045.1_ASM130404v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001304045.1_ASM130404v1_protein.faa.gz 2091 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2092 sequences in 2018-test_datasets/bacteroides/protein/GCA_001304045.1_ASM130404v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00267 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001304045.1_ASM130404v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001310955.1_ASM131095v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001310955.1_ASM131095v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001310955.1_ASM131095v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001373135.1_2e6A_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001373135.1_2e6A_assembly_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001373135.1_2e6A_assembly_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001398875.1_Mother10-2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001398875.1_Mother10-2_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001398875.1_Mother10-2_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001405895.1_14207_7_22_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001405895.1_14207_7_22_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001405895.1_14207_7_22_protein.faa.gz 5035 sequences\r",
      "\u001b[Kcalculated 6 signatures for 5036 sequences in 2018-test_datasets/bacteroides/protein/GCA_001405895.1_14207_7_22_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00179 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001405895.1_14207_7_22_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001405935.1_13414_6_56_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001405935.1_13414_6_56_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001405935.1_13414_6_56_protein.faa.gz 4271 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4272 sequences in 2018-test_datasets/bacteroides/protein/GCA_001405935.1_13414_6_56_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00152 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001405935.1_13414_6_56_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001413955.1_ASM141395v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001413955.1_ASM141395v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001413955.1_ASM141395v1_protein.faa.gz 3384 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3385 sequences in 2018-test_datasets/bacteroides/protein/GCA_001413955.1_ASM141395v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00180 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001413955.1_ASM141395v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001443605.1_ASM144360v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001443605.1_ASM144360v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001443605.1_ASM144360v1_protein.faa.gz 3864 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3865 sequences in 2018-test_datasets/bacteroides/protein/GCA_001443605.1_ASM144360v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00158 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001443605.1_ASM144360v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001548195.1_ASM154819v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001548195.1_ASM154819v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001548195.1_ASM154819v1_protein.faa.gz 2509 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2510 sequences in 2018-test_datasets/bacteroides/protein/GCA_001548195.1_ASM154819v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00298 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001548195.1_ASM154819v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001578565.1_ASM157856v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001578565.1_ASM157856v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001578565.1_ASM157856v1_protein.faa.gz 5104 sequences\r",
      "\u001b[Kcalculated 6 signatures for 5105 sequences in 2018-test_datasets/bacteroides/protein/GCA_001578565.1_ASM157856v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00203 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001578565.1_ASM157856v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001595155.1_CFSAN033860_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001595155.1_CFSAN033860_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001595155.1_CFSAN033860_01.0_protein.faa.gz 4762 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4763 sequences in 2018-test_datasets/bacteroides/protein/GCA_001595155.1_CFSAN033860_01.0_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00151 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001595155.1_CFSAN033860_01.0_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001603535.1_ASM160353v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001603535.1_ASM160353v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001603535.1_ASM160353v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001603695.1_ASM160369v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001603695.1_ASM160369v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001603695.1_ASM160369v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001614375.1_ASM161437v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001614375.1_ASM161437v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001614375.1_ASM161437v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001659685.2_ASM165968v2_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001659685.2_ASM165968v2_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_001659685.2_ASM165968v2_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001693695.1_ASM169369v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001693695.1_ASM169369v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001693695.1_ASM169369v1_protein.faa.gz 4069 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4070 sequences in 2018-test_datasets/bacteroides/protein/GCA_001693695.1_ASM169369v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00167 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001693695.1_ASM169369v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001761325.1_ASM176132v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001761325.1_ASM176132v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001761325.1_ASM176132v1_protein.faa.gz 2969 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2970 sequences in 2018-test_datasets/bacteroides/protein/GCA_001761325.1_ASM176132v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00173 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001761325.1_ASM176132v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001816245.1_ASM181624v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001816245.1_ASM181624v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001816245.1_ASM181624v1_protein.faa.gz 4955 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4956 sequences in 2018-test_datasets/bacteroides/protein/GCA_001816245.1_ASM181624v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00208 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001816245.1_ASM181624v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_001915365.1_ASM191536v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_001915365.1_ASM191536v1_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_001915365.1_ASM191536v1_protein.faa.gz 3280 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3281 sequences in 2018-test_datasets/bacteroides/protein/GCA_001915365.1_ASM191536v1_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00169 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_001915365.1_ASM191536v1_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_002036985.1_ASM203698v1_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_002036985.1_ASM203698v1_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_002036985.1_ASM203698v1_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900032635.1_14555_6_27_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900032635.1_14555_6_27_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900032635.1_14555_6_27_protein.faa.gz 2029 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2030 sequences in 2018-test_datasets/bacteroides/protein/GCA_900032635.1_14555_6_27_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00191 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900032635.1_14555_6_27_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900104585.1_PRJEB16348_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900104585.1_PRJEB16348_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_900104585.1_PRJEB16348_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900106755.1_IMG-taxon_2693429916_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900106755.1_IMG-taxon_2693429916_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900106755.1_IMG-taxon_2693429916_annotated_assembly_protein.faa.gz 4749 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4750 sequences in 2018-test_datasets/bacteroides/protein/GCA_900106755.1_IMG-taxon_2693429916_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00151 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900106755.1_IMG-taxon_2693429916_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900109385.1_IMG-taxon_2593339226_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900109385.1_IMG-taxon_2593339226_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900109385.1_IMG-taxon_2593339226_annotated_assembly_protein.faa.gz 4846 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4847 sequences in 2018-test_datasets/bacteroides/protein/GCA_900109385.1_IMG-taxon_2593339226_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00160 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900109385.1_IMG-taxon_2593339226_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900110645.1_IMG-taxon_2693429910_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900110645.1_IMG-taxon_2693429910_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900110645.1_IMG-taxon_2693429910_annotated_assembly_protein.faa.gz 4891 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4892 sequences in 2018-test_datasets/bacteroides/protein/GCA_900110645.1_IMG-taxon_2693429910_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00158 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900110645.1_IMG-taxon_2693429910_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900111425.1_IMG-taxon_2693429902_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900111425.1_IMG-taxon_2693429902_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900111425.1_IMG-taxon_2693429902_annotated_assembly_protein.faa.gz 4089 sequences\r",
      "\u001b[Kcalculated 6 signatures for 4090 sequences in 2018-test_datasets/bacteroides/protein/GCA_900111425.1_IMG-taxon_2693429902_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00170 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900111425.1_IMG-taxon_2693429902_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900112795.1_IMG-taxon_2693429893_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900112795.1_IMG-taxon_2693429893_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900112795.1_IMG-taxon_2693429893_annotated_assembly_protein.faa.gz 3056 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3057 sequences in 2018-test_datasets/bacteroides/protein/GCA_900112795.1_IMG-taxon_2693429893_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00234 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900112795.1_IMG-taxon_2693429893_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900116355.1_IMG-taxon_2622736518_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900116355.1_IMG-taxon_2622736518_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900116355.1_IMG-taxon_2622736518_annotated_assembly_protein.faa.gz 5086 sequences\r",
      "\u001b[Kcalculated 6 signatures for 5087 sequences in 2018-test_datasets/bacteroides/protein/GCA_900116355.1_IMG-taxon_2622736518_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00153 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900116355.1_IMG-taxon_2622736518_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900128475.1_PRJEB18047_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900128475.1_PRJEB18047_protein.faa.gz\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/bin/sourmash\", line 11, in <module>\n",
      "    load_entry_point('sourmash', 'console_scripts', 'sourmash')()\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/__main__.py\", line 83, in main\n",
      "    cmd(sys.argv[2:])\n",
      "  File \"/Users/tessa/Dropbox/dib-lab/2019-fall-rotation-diblab/hp_encoding/czbiohub_sourmash/sourmash/command_compute.py\", line 335, in compute\n",
      "    for n, record in enumerate(screed.open(filename)):\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 39, in __init__\n",
      "    self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "  File \"/Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/screed/openscreed.py\", line 95, in open_reader\n",
      "    raise ValueError(\"unknown file format for '%s'\" % filename)\n",
      "ValueError: unknown file format for '2018-test_datasets/bacteroides/protein/GCA_900128475.1_PRJEB18047_protein.faa.gz'\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900142325.1_IMG-taxon_2698536696_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900142325.1_IMG-taxon_2698536696_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900142325.1_IMG-taxon_2698536696_annotated_assembly_protein.faa.gz 3334 sequences\r",
      "\u001b[Kcalculated 6 signatures for 3335 sequences in 2018-test_datasets/bacteroides/protein/GCA_900142325.1_IMG-taxon_2698536696_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00209 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900142325.1_IMG-taxon_2698536696_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "\r",
      "\u001b[K== This is sourmash version 2.0.0a10.dev125+gbc9a050. ==\n",
      "\r",
      "\u001b[K== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==\n",
      "\n",
      "\r",
      "\u001b[KWARNING: input is protein, turning off nucleotide hashing\n",
      "\r",
      "\u001b[Kcomputing signatures for files: 2018-test_datasets/bacteroides/protein/GCA_900167375.1_IMG-taxon_2588254259_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[KComputing signature for ksizes: [7, 11, 17]\n",
      "\r",
      "\u001b[KComputing only protein (and not nucleotide) signatures.\n",
      "\r",
      "\u001b[KComputing a total of 3 signature(s).\n",
      "\r",
      "\u001b[KTracking abundance of input k-mers.\n",
      "\r",
      "\u001b[K... reading sequences from 2018-test_datasets/bacteroides/protein/GCA_900167375.1_IMG-taxon_2588254259_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[K...2018-test_datasets/bacteroides/protein/GCA_900167375.1_IMG-taxon_2588254259_annotated_assembly_protein.faa.gz 2332 sequences\r",
      "\u001b[Kcalculated 6 signatures for 2333 sequences in 2018-test_datasets/bacteroides/protein/GCA_900167375.1_IMG-taxon_2588254259_annotated_assembly_protein.faa.gz\n",
      "\r",
      "\u001b[Ktime taken to save signatures is 0.00154 seconds\r",
      "\r",
      "\u001b[Ksaved signature(s) to sigs/bacteroides/protein/hp/GCA_900167375.1_IMG-taxon_2588254259_annotated_assembly_protein.sig. Note: signature license is CC0.\n",
      "usage: sourmash [-o OUTPUT] [--ignore-abundance] [-k KSIZE] [--protein]\n",
      "                [--no-protein] [--dayhoff] [--no-dayhoff] [--hp] [--no-hp]\n",
      "                [--dna] [--no-dna] [--traverse-directory] [--csv CSV]\n",
      "                [-p PROCESSES] [-q]\n",
      "                signatures [signatures ...]\n",
      "sourmash: error: argument --csv: can't open 'sourmash_compare/bacteroides_k7_prot_hp_comp.csv': [Errno 2] No such file or directory: 'sourmash_compare/bacteroides_k7_prot_hp_comp.csv'\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'# ^this is magic code telling jupyter that this block is entirely bash code.\\n\\n#For example, you can run:\\n\\nmkdir -p sigs/bacteroides/protein/hp\\n# generate hp signatures for bacteroides\\nfor infile in 2018-test_datasets/bacteroides/protein/*faa.gz; do out_name=$(basename $infile .faa.gz); sourmash compute -k 7,11,17 --input-is-protein --hp --track-abundance -o sigs/bacteroides/protein/hp/${out_name}.sig ${infile}; done\\n# and build the compare matrix for k=7:\\nsourmash compare -k 7 --csv sourmash_compare/bacteroides_k7_prot_hp_comp.csv sigs/bacteroides/protein/hp/*.sig\\n'' returned non-zero exit status 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-98802e6d72bc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'# ^this is magic code telling jupyter that this block is entirely bash code.\\n\\n#For example, you can run:\\n\\nmkdir -p sigs/bacteroides/protein/hp\\n# generate hp signatures for bacteroides\\nfor infile in 2018-test_datasets/bacteroides/protein/*faa.gz; do out_name=$(basename $infile .faa.gz); sourmash compute -k 7,11,17 --input-is-protein --hp --track-abundance -o sigs/bacteroides/protein/hp/${out_name}.sig ${infile}; done\\n# and build the compare matrix for k=7:\\nsourmash compare -k 7 --csv sourmash_compare/bacteroides_k7_prot_hp_comp.csv sigs/bacteroides/protein/hp/*.sig\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2357\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2358\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2359\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2360\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2361\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m</Users/tessa/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/decorator.py:decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/sourmash-hp/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'# ^this is magic code telling jupyter that this block is entirely bash code.\\n\\n#For example, you can run:\\n\\nmkdir -p sigs/bacteroides/protein/hp\\n# generate hp signatures for bacteroides\\nfor infile in 2018-test_datasets/bacteroides/protein/*faa.gz; do out_name=$(basename $infile .faa.gz); sourmash compute -k 7,11,17 --input-is-protein --hp --track-abundance -o sigs/bacteroides/protein/hp/${out_name}.sig ${infile}; done\\n# and build the compare matrix for k=7:\\nsourmash compare -k 7 --csv sourmash_compare/bacteroides_k7_prot_hp_comp.csv sigs/bacteroides/protein/hp/*.sig\\n'' returned non-zero exit status 2."
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "# ^this is magic code telling jupyter that this block is entirely bash code.\n",
    "\n",
    "mkdir -p sigs/bacteroides/protein/hp\n",
    "# generate hp signatures for bacteroides\n",
    "for infile in 2018-test_datasets/bacteroides/protein/*faa.gz; do out_name=$(basename $infile .faa.gz); sourmash compute -k 7,11,17 --input-is-protein --hp --track-abundance -o sigs/bacteroides/protein/hp/${out_name}.sig ${infile}; done\n",
    "# and build the compare matrix for k=7:\n",
    "sourmash compare -k 7 --csv sourmash_compare/bacteroides_k7_prot_hp_comp.csv sigs/bacteroides/protein/hp/*.sig\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
